operator/internal/manifests/internal/rules/marshal.go

```diff
@@ -20,6 +22,16 @@ func MarshalAlertingRule(a lokiv1.AlertingRule) (string, error) {
 		Groups: a.Spec.Groups,
 	}
 
+	for _, group := range ar.Groups {
+		for _, rule := range group.Rules {
+			if rule.Labels == nil {
+				rule.Labels = map[string]string{}
+			}
+
+			rule.Labels[tenantLabel] = a.Spec.TenantID
```

Comment:
Sorry, I thought I had posted this comment last week already, but apparently I never clicked "publish" ...

It looks like we need to do a `DeepCopy` somewhere, because we are modifying the `Groups` (dirtying the lister cache). These are all pointer-types, so not only the one in `alertingRuleSpec` is modified but also the original in `AlertingRule`.

Reply:
Yeah you posted this on slack, I started working it out and stashed it. It got forgotten :(